crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -58,10 +63,164 @@ impl RequestPump {
                 _ = &mut cancel => {
                     break;
                 }
-                Some(_append) = self.append_stream.next() => {
+                Some(append) = self.append_stream.next() => {
+                    self.handle_append(Arc::clone(&provider), append).await;
                 }
             }
         }
+
+        Ok(())
+    }
+
+    /// Infailable handle_append method
+    async fn handle_append<T: TransportConnect>(
+        &mut self,
+        provider: Arc<ReplicatedLogletProvider<T>>,
+        incoming: Incoming<Append>,
+    ) {
+        let task = HandleAppendTask {
+            metadata: self.metadata.clone(),
+            provider,
+        };
+
+        // we spawn immediately to avoid slowing the request pump loop
+        // in case responses will go over a slow channel
+        let _ = task_center().spawn_child(
+            TaskKind::Disposable,
+            "handle-append-request",
+            None,
+            task.run(incoming),
+        );
+    }
```

Comment:
How do we uphold the original order of appends from the same sender with this approach? This appears to violate the ordering guarantees if we run tasks for appends coming from the same producer concurrently.

Reply:
Hmm, very good point. Before I refactor I actually only spawned after the records has already been queued. Thanks for catching this